### ISSUE TITLE
Casting conflict with `IN` clause & enums for raw queries

### ISSUE THREAD

**Issue by Sytten** [prisma/prisma#14956]
### Bug description

With the [upgrade of raw queries in Prisma v4](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-4#raw-query-mapping-postgresql-type-casts), you now must types enums to their "real" type. This is not playing very well with sets and you have to use an array instead. I am not sure if there is something actionnable on prisma side (except more documentation for sure), but it could be nice to have more helpers methods for that use-case.

### How to reproduce

1. Given a prisma enum `OrderStatus` and `const SOLD_STATUSES = Array<OrderStatus>(OrderStatus.PROCESSING, ...);`
2. Working where clause before upgrade: `"order".status IN (${Prisma.join(SOLD_STATUSES)})`
3. Must be changed to either:
    - `"order".status = ANY (ARRAY[${Prisma.join(SOLD_STATUSES)}]::order_status[])`
    - `"order".status = ANY (${SOLD_STATUSES}::order_status[])`

### Expected behavior

Ideally you can use the newer `IN` syntax instead of relying on an array.

### Prisma information

See above

### Environment & setup

- OS: MacOS
- Database: Postgres
- Node.js version: 18

### Prisma Version

```
Version: 4.2.1
```

**Comment by floriantz** [comment]
I have a similar issue https://github.com/prisma/prisma/issues/14978 with a simpler case with a uuid type cast.

**Comment by Sytten** [comment]
Yeah I feel it could be nice if the casting was integrated in prisma directly.
I could see things like: `Prisma.cast("expression", "type")` and `Prisma.in(MY_ARRAY, "type")`

**Comment by ludralph** [comment]
Closing this in favour of [14978](https://github.com/prisma/prisma/issues/14978)
